asmcli/asmcli.sh

```diff
@@ -70,13 +70,16 @@ RELEASE_LINE=""
 PREVIOUS_RELEASE_LINE=""
 KPT_BRANCH=""
 NAMESPACE_EXISTS=0
+KUBECONFIG_SUPPLIED=0
 
 ### Option variables ###
 PROJECT_ID="${PROJECT_ID:=}"
 CLUSTER_NAME="${CLUSTER_NAME:=}"
 CLUSTER_LOCATION="${CLUSTER_LOCATION:=}"
 MODE="${MODE:=}"
 CA="${CA:=}"
+KUBECONFIG="${KUBECONFIG:=}"
```

Comment:
I'm slightly concerned about the variable name here. I think that KUBECONFIG is something that's probably set and exported in shells pretty regularly, which makes me worry that customers running the script might inadvertently pass this without realizing it.

If you don't think it's an issue, feel free to ignore. Otherwise maybe KUBECONFIG_FILE or something is a safer name for the variable/flag?

Reply:
we need it to avoid regenerating the credentials inside **validate_cluster** and **configure_kubectl** because **set_up_local_workspace** export a _KUBECONFIG_ before they run.

Reply:
That makes sense. We can definitely have an internal `KUBECONFIG` variable, but maybe we should explicitly set it to an empty string instead of setting it to the existing value. (maybe just `KUBECONFIG="${KUBECONFIG_FILE:=}"`?)

Reply:
sorry i misunderstood the earlier comment. i agree with the concern and will update it to _KUBECONFIG_FILE_

Reply:
done, changed from **KUBECONFIG** -> **KUBECONFIG_FILE**